### DISCUSSION TITLE
SQL ETLs using custom IDs

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#16883] [Q&A]
I have a single document that contains a bunch of various settings. What I'd like to do is loop over every setting in the document and push that into a SQL database. The SQL schema has 3 fields - A client Id (Guid), a Key, and a Value;

Given the Settings document (which contains other information beside the settings themselves), I'd like to loop over each setting object in SettingsDocument.Settings and move them to SQL.

In the ETL, I set the SQL Table to ClientSetting but the Document ID Column isn't what I want for the ID. I want the ID of each setting in the document to be the Key.

I'm trying this 

this.Settings.forEach(s => {
     loadToClientSetting({
        Key: s.Id,
        ClientId: "84bb9307-9a70-4b14-8898-b33bd1f30af0",
        Value: s.Value
    });   
});

where s.Id represents the Id of the individual setting, but it's always setting that as the _document.Id_. Is there a way to use something else as the identifier?

**Comment by ayende** [comment]
For SQL ETL, there must be _a_ column that would contain the document id.
It doesn't have to be the key column.

That is so we'll be able to update the value on document update.
We cannot use `ClientId`, since how you'll handle a removal of a value.
Instead, we first send a `delete from $table where $doc_id_col = $doc_id` and then insert the new values.

I would recommend adding another column here to hold the document id (if lots of rows in the table, make sure it is indexed)
